File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    pattern = [[1, 2, 3, 4, 5], [2, 1, 2, 3, 2, 4, 2, 5],
           [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]]

    score = [0] * 3

    for i in range(len(answers)):
        if pattern[0][i%len(pattern[0])] == answers[i]:
            score[0] += 1
        if pattern[1][i%len(pattern[1])] == answers[i]:
            score[1] += 1
        if pattern[2][i%len(pattern[2])] == answers[i]:
            score[2] += 1
    
    high = max(score)
    a = []
    for i in range(len(score)):
        if score[i] == high:
            a.append(i+1)
    return a

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int [][] pattern = {
                {1,2,3,4,5},
                {2,1,2,3,2,4,2,5},
                {3,3,1,1,2,2,4,4,5,5}
        };

        int[] score = new int[3];
        List<Integer> list = new ArrayList<Integer>();

        for (int i = 0; i < answers.length; i++) {
            if (pattern[0][i%pattern[0].length] == answers[i]) {
                score[0]++;
            }
            if (pattern[1][i%pattern[1].length] == answers[i]) {
                score[1]++;
            }
            if (pattern[2][i%pattern[2].length] == answers[i]) {
                score[2]++;
            }
        }

        int maxScore = Arrays.stream(score).max().getAsInt();

        for (int i = 0; i < 3; i++) {
            if (maxScore == score[i]) {
                list.add(i+1);
            }
        }
        return list.stream().mapToInt(i -> i).toArray();
    }
}